pkgs/applications/virtualization/qemu/default.nix

```
@@ -35,6 +35,21 @@ let
   audio = lib.optionalString alsaSupport "alsa,"
     + lib.optionalString pulseSupport "pa,"
     + lib.optionalString sdlSupport "sdl,";
+  emit-kvm-warnings = writeShellScriptBin "emit-kvm-warnings" ''
+    WARNCOL='\033[0m'
+    NEUTRALCOL='\033[0m'
+    WARNING="''${WARNCOL}warning:''${NEUTRALCOL}"
+    if [ ! -e /dev/kvm ]; then
+      echo -e "''${WARNING} KVM is not installed - execution will be slow"
+      echo "Consider installing KVM for hardware-accelerated execution."
+    elif [ ! -r /dev/kvm ] || [ ! -w /dev/kvm ]; then
```

Nitpick: no need for two invocations of `[` here.

```suggestion
    elif [ ! -r /dev/kvm -o ! -w /dev/kvm ]; then
```

---

Done in d033935ab6fd850160e31a1db822486f39650210